### PR TITLE
Lets you tip over medbots (you monster)

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
[](https://i.imgur.com/cT006pM.png)

A while ago I read a book about people and robots and emotions, _Alone Together_ by Sherry Turkle, and there's one passage that talks about people's reactions to robots simulating distress, specifically with how Furbies cry when you hold them upside down.

> A person is asked to invert three creatures: a Barbie doll, a Furby, and a biological gerbil. [Freedom] Baird's question is simple: "How long can you hold the object upside down before your emotions make you turn it back?"

> People are surprised by how upset they get in this theater of distress. And then they get upset that they are upset. They often try to reassure themselves, saying things like, “Chill, chill, it’s only a toy!” They are experiencing something new: you can feel bad about yourself for how you behave with a computer program. Adults come to the upside-down test knowing two things: the Furby is a machine and they are not torturers. By the end, with a whimpering Furby in tow, they are on new ethical terrain.

Ethics memes aside, the idea off people being caught off guard at their guilt for causing harm to inanimate objects is really funny to me, and feels extra relevant to SS13. Considering people already regularly get in fights and maim each other over harming station pets, I picture this being a great source of arguments and fights ICly over what torturing a cute lil' medbot says about you.

tl;dr this PR lets you tip over medbots which prevents them from moving or healing, and they'll slowly get more and more distressed and desperate asking for help from people around them. After a few minutes, if no one helps right them, they'll manage to right themselves up and snitch on whoever tipped them over to medical so the psychologist can keep an eye on them. 

[](https://i.imgur.com/c1Lvyvu.png)

## Why It's Good For The Game
Gives the psychologist more ammo to call people crazy, gives bored assistants another way to seek attention, provides source of minor IC conflict

## Changelog
:cl: Ryll/Shaps
add: You can now tip over medbots! 
/:cl:

I'd like to get voicelines for these as well, but I wasn't sure of a good TTS generator, and also wanted to see if people think this is worth doing before devoting too much time to it.

### CODE REVIEW

**Comment by Doosteroni** [comment]
Nice

**Comment by PKPenguin321** [comment]
So stunning them makes you insane but murdering them for parts is okay. I'm in the clear, phew

**Comment by Ryll-Ryll** [comment]
If I can trick people into actually arguing about the ethics of medbot abuse and whether it's morally justifiable to kill one for parts, I'll die happy

**Comment by ArcaneDefence** [comment]
Truly the medbot nerf we needed.

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 1]
```diff
@@ -226,10 +236,75 @@
 	else
 		return
 
+/mob/living/simple_animal/bot/medbot/proc/tip_over(mob/user)
+	playsound(src, 'sound/machines/warning-buzzer.ogg', 50)
+	user.visible_message("<span class='danger'>[user] tips over [src]!</span>", "<span class='danger'>You tip [src] over!</span>")
+	mode = BOT_TIPPED
+	var/matrix/mat = transform
+	transform = mat.Turn(180)
+	tipper_name = user.name
```
Comment (Qustinnus) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:245]: I don't think this'll look good in a message if the person has a mask

Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:245]: It gets whatever their visible name is at the time of tipping, I figured it'd be more immersion breaking if the medbot magically deduced your identity than it would be to call out that some unknown was bullying them. Plus this lets you disguise as someone else to tip the bot so that the other person gets reported for being a psycho

Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:245]: I agree, I was more worried about them calling you out on a name like "john doe (disguised as bob stink)"

Comment (Ryll-Ryll) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:245]: If it's good enough for beepsky it's good enough for medbots, I'm pretty sure I do it the same way for the Orion Trail snitching too

Comment (Qustinnus) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:245]: I guess so, lets hope it doesnt cause any weird behavior :)

**Comment by trollbreeder** [comment]
Can we also tip over cyborgs?

**Comment by Ryll-Ryll** [comment]
Honestly i think people would sooner empathize with a brainless medbot than they would another player, especially a silicon player.

**Comment by wesoda25** [comment]
It'd be pretty cool if gravity stuff/explosions had a chance to tip them over

**Comment by mrhugo13** [comment]
Man, what the fuck did medbots ever do to you?

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 2]
```diff
@@ -226,10 +236,75 @@
 	else
 		return
 
+/mob/living/simple_animal/bot/medbot/proc/tip_over(mob/user)
+	playsound(src, 'sound/machines/warning-buzzer.ogg', 50)
+	user.visible_message("<span class='danger'>[user] tips over [src]!</span>", "<span class='danger'>You tip [src] over!</span>")
+	mode = BOT_TIPPED
+	var/matrix/mat = transform
+	transform = mat.Turn(180)
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:244]: would this look better if the medibot was only rotated by 90 degrees?

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 3]
```diff
@@ -341,6 +416,17 @@
 	if(C.getToxLoss() >= heal_threshold)
 		return TRUE
 
+/mob/living/simple_animal/bot/medbot/attack_hand(mob/living/carbon/human/H)
+	if(H.a_intent == INTENT_DISARM && mode != BOT_TIPPED)
+		H.visible_message("<span class='danger'>[H] begins tipping over [src].</span>", "<span class='danger'>You begin tipping over [src]...</span>")
+		speak(pick(list("Hey, wait...", "Please don't...", "Whoa, hold on-", "Wait!-", "I trusted you!")))
+		if(do_after(H, 3 SECONDS, target=src))
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:423]: this do_after makes it kind of hard to tip over a medibot that's on patrol, but at the same time, being able to instantly tip over pAI medibots would be kind of lame on the pAI medibot's end.

maybe remove this do_after() check for NPC medibots, but make medibots that have minds require the do_after to be tipped over?

actually, how does this whole medibot panic system interact with player-controlled medibots? can they ever right themselves? does being tipped over even AFFECT a player-controlled medibot, or can you just roam around as an Australian medibot?

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 4]
```diff
@@ -226,10 +236,75 @@
...
+		if(MEDBOT_PANIC_MED)
+			speak("Hey, I need help! Please help me up!")
+		if(MEDBOT_PANIC_HIGH)
+			speak("Please help! This is scaring me!!")
+		if(MEDBOT_PANIC_FUCK)
+			speak("I DON'T LIKE BEING TIPPED! PLEASE HELP ME!!")
+		if(MEDBOT_PANIC_ENDING)
+			speak("THIS HURTS, PLEASE HELP ME!!")
+		if(MEDBOT_PANIC_END)
+			speak("PSYCH ALERT: Crewmember [tipper_name] recorded displaying antisocial tendencies torturing bots in [get_area(src)]. Please schedule psych evaluation.", radio_channel)
```
Comment (ATH1909) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:274]: it's a bit weird that the medibot immediately goes from screaming in panic to totally fine (and even calmly making a psych report about its traumatic experience) immediately after it rights itself.

also, you might want to add a word between "tendencies" and "torturing".

**Comment by ATH1909** [comment]
amazing PR, 10/10

File: code/modules/mob/living/simple_animal/bot/medbot.dm (hunk 2 above)
Comment (trollbreeder) [reply code/modules/mob/living/simple_animal/bot/medbot.dm:244]: yes it looks a bit better if it's 90 degrees

**Comment by trollbreeder** [comment]
image

**Comment by Fikou** [comment]
why not make tipping something more generalized? cows can already be tipped

**Comment by Ryll-Ryll** [comment]
Because medbots are uniquely pure and are canonically the ss13 equivalent of furbies. That's right, you heard it here first, medbot designs are descendents of furbies

File: code/modules/mob/living/simple_animal/bot/medbot.dm [hunk 5]
```diff
@@ -226,10 +236,75 @@
...
+		visible_message("<span class='notice'>[src] manages to writhe around enough to right itself.</span>")
+		tipper_name = null
+	tipped_status = MEDBOT_PANIC_NONE
+	mode = BOT_IDLE
+	transform = matrix()
+
+/// if someone tipped us over, check whether we should ask for help or just right ourselves eventually
+/mob/living/simple_animal/bot/medbot/proc/handle_panic()
+	tipped_status++
+	switch(tipped_status)
```
Comment (spessbandit) [inline code/modules/mob/living/simple_animal/bot/medbot.dm:262]: I love this ID but the dialogue here seems like it could be workshopped to be more funny

**Comment by tmtmtl30** [comment]
can't believe that when *i* make a PR to let players RESPECT medibots, it gets DENIED, but when RYLL makes a PR to let players BULLY medibots, it gets PRAISED. really says a lot about the state of the codebase!

(👍)

**Comment by ATH1909** [comment]
> can't believe that when _i_ make a PR to let players RESPECT medibots, it gets DENIED, but when RYLL makes a PR to let players BULLY medibots, it gets PRAISED. really says a lot about the state of the codebase!
> 
> (👍)

can you post a link to that medibot-respecting PR?

**Comment by Shadowflame909** [comment]
>trollbreeder

New meta for destroying sec borgs and syndie borgs. It is time to make them common

Good PR though. Love it.

**Comment by tmtmtl30** [comment]
@ATH1909 https://github.com/tgstation/tgstation/pull/46957

**Comment by genessee596** [comment]
Good PR, hope to see it get merged!

**Comment by Ziiro** [comment]
FYI The voice is Dr. Sbaitso. See below for how to make new clips.

https://www.reddit.com/r/SS13/comments/7ne0mg/medibot_texttospeech_question/

Also I tipped a medibot and I felt like a fucking monster, well done.
